ONSdigital/census-rm-pubsub
# Receipt with qid

## Motivation and Context
receipting currently fails as the case_id is assumed mandatory but it isn't, instead we need to receipt using QID (the case-processor expects this to always exist in the rabbit message) and add case_id to the rabbit message if its passed to us

## What has changed
updated subscriber.py to pull out "questionnaire_id" out of the pubsub payload (and case_id if its there) and add them to the rabbit message 

## How to test?
- run acceptance tests
- load a sample into your GCP project, pick a case and get the case-id and qid and run the following command to simulate an eq submission
`gsutil -h x-goog-meta-questionnaire_id:<qid> -h x-goog-meta-tx_id:123456789 -h x-goog-meta-case_id:f<case-id> cp 123456789 gs://receipting-bucket-<project-name>`
you should now see that case being receipted correctly (new receipt event logged, receipt_received flat set to true)
